packages/xo-server/src/api/xostor.mjs

```diff
@@ -212,6 +212,8 @@ export const create = defer(async function (
 
     const host = hosts[0]
     const xapi = this.getXapi(host)
+    const poolHostIds = Object.keys(xapi.objects.indexes.type.host)
+    const poolHosts = poolHostIds.map(id => this.getObject(id, 'host'))
```

Comment:
```js
const poolHosts = xapi.objects.indexes.type.host
// ...
await handleHostsDependencies(Object.values(poolHosts))
```

Reply:
`xapi.objects.indexes.type.host` returns `XAPI objects`, but `handleHostsDependencies` requires `XO objects` because it uses `obj._xapiRef`

Reply:
In theory, API files should simply be the interface with users, the logic should be in a dedicated XO mixins.

Also, it's better to manage XAPI objects directly instead of relying on XO objects, that's what I've been doing for all the recent code.

Not asking you to change anything, just saying for the future.